source/server/config_validation/server.h

```diff
@@ -71,7 +70,9 @@ class ValidationInstance final : Logger::Loggable<Logger::Id::main>,
                      Filesystem::Instance& file_system);
 
   // Server::Instance
-  Admin& admin() override { return *admin_; }
+  OptRef<Admin> admin() override {
+    return makeOptRefFromPtr(reinterpret_cast<Envoy::Server::Admin*>(admin_.get()));
```

Comment:
I'm surprised to see reinterpret_cast here instead of static_cast. Does static_cast work here?